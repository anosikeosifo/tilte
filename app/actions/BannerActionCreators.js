import constants from '../constants';

export default class BannerActionCreators {
  static toggleNext() {
    return ({
      type: constants.BANNER_TOGGLE_NEXT,
    });
  }

  static togglePrevious() {
    return ({
      type: constants.BANNER_TOGGLE_PREV,
    });
  }
}
