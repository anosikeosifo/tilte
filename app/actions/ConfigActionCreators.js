import constants from '../constants';
import ConfigAPI from '../api/ConfigAPI';

export default class ConfigActionCreators {
  static getAppData() {
    return (dispatch) => {
      dispatch({ type: constants.FETCH_APP_DATA });
    }
  }
  static fetchConfigData() {
    return (dispatch) => {
      dispatch({ type: constants.FETCH_CONFIG_DATA });
      ConfigAPI.fetchConfig().then(
        (config) => dispatch({ type: constants.RECEIVE_CONFIG_DATA, success: true, payload: config }),
        (error) => dispatch({ type: constants.RECEIVE_CONFIG_DATA, success: false, error })
      );
    }
  }

  static updateCategoryPreference() {
    return (dispatch) => {
      dispatch({ type: constants.UPDATE_CATEGORY_CONFIG });
    }
  }
}
