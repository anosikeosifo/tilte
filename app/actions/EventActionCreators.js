import { EVENT_ACTIONS } from '../constants';
import { fetchEventDetailsAPI, fetchTrendingEventsAPI, fetchFeaturedEventsAPI,loadMapAPI, eventRegistrationAPI, saveEventAPI } from '../api/EventAPI';
import { buildPostParams, buildUrlQueryParams } from '../helpers/HttpHelper';

const eventActionSuccess = (actionType, payload) => ({
  type: EVENT_ACTIONS[`${actionType}Success`], success: true, payload
});

const eventActionError = (actionType, error) => ({
  type: EVENT_ACTIONS[`${actionType}Error`], success: false, error
});

const triggerlikeEvent = () => ({
  type: EVENT_ACTIONS.likeRequest
});

const triggerFetchEventDetails = () => ({
  type: EVENT_ACTIONS.fetchDetails
});

const triggerFetchComments = () => ({
  type: EVENT_ACTIONS.fetchCommentsRequest
});

const triggerPostComment = () => ({
  type: EVENT_ACTIONS.postCommentRequest
});

const triggerLoadMap = () => ({
  type: EVENT_ACTIONS.loadMapRequest
});

const triggerEventRegistration = () => ({
  type: EVENT_ACTIONS.eventRegistrationRequest
});

const triggerSaveEvent = () => ({
  type: EVENT_ACTIONS.saveEventRequest
});

const triggerFetchFeatured = () =>({
  type: EVENT_ACTIONS.fetchFeaturedRequest
});

const triggerFetchTrending = () =>({
  type: EVENT_ACTIONS.ftriggerFetchTrendingRequest
});

export const likeEvent = (eventId, actorId) => {
  const actionType = 'like';
  const requestParams = buildPostParams({
    event_id: eventId, user_id: actorId
  });

  return (dispatch) => {
    triggerlikeEvent();

    likeEventAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      .catch(error => dispatch(eventActionError(actionType, error)));
  }
};

export const fetchEventDetails = (eventId, actorId) => {
  const actionType = 'fetchDetails';
  const requestParams = buildPostParams({
    event_id: eventId, user_id: actorId
  });

  return (dispatch) => {
    triggerFetchEventDetails();

    //this should take requestParams as arg
    fetchEventDetailsAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(actionType, error)));
  }
};

export const fetchComments = (eventId, actorId) => {
  const actionType = 'fetchComments';
  const requestParams = {
    event_id: eventId, user_id: actorId, page: '1'
  };

  return (dispatch) => {
    triggerFetchComments();
    fetchCommentsAPI (requestParams)
      .then(payload => eventActionSuccess(actionType, payload))
      .catch(error => eventActionError(actionType, error));
  };
}

export const postComment = (eventId, actorId, content) => {
  const actionType = 'postComment';
  const requestParams = buildPostParams({
    event_id: eventId,
    user_id: actorId,
    text: content,
  });

  return (dispatch) => {
    triggerPostComment();
    postCommentAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(error)));
  };
};

export const loadMapView = (eventId, actorId, longitude=null, latitude=null) => {
  const actionType = 'loadMap';
  const requestParams = buildPostParams({
    event_id: eventId,
    user_id: actorId,
    longitude,
    latitude,
  });

  return (dispatch) => {
    triggerLoadMap();
    loadMapAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(error)));
  };
}

export const registerForEvent = (eventId, actorId, longitude=null, latitude=null) => {
  const actionType = 'eventRegistration';
  const requestParams = buildPostParams({
    event_id: eventId,
    user_id: actorId,
    longitude,
    latitude,
  });

  return (dispatch) => {
    triggerEventRegistration();
    eventRegistrationAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(error)));
  };
}

export const saveEvent = (eventId, actorId) => {
  const actionType = 'saveEvent';
  const requestParams = buildPostParams({
    event_id: eventId,
    user_id: actorId,
  });

  return (dispatch) => {
    triggerSaveEvent();
    saveEventAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(error)));
  };
}

export const fetchTrendingEvents = (userId) => {
  const actionType = 'fetchTrending';
  const requestParams = buildPostParams({ user_id: userId });

  return (dispatch) => {
    triggerFetchTrending();
    fetchTrendingEventsAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(error)));
  };
}

export const fetchFeaturedEvents = (actorId, locationLat, locationLong) => {
  const actionType = 'fetchFeatured';
  const requestParams = buildPostParams({
    user_id: actorId,
    longitude: locationLong,
    latitude: locationLat,
  });

  return (dispatch) => {
    triggerFetchFeatured();
    fetchFeaturedEventsAPI()
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(error)));
  };
}
