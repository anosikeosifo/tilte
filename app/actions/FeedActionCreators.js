import constants from '../constants';
import FeedAPI from '../api/FeedAPI';

export default class FeedActionCreators {
  static fetchUserFeed(userId, userLocation, pageNumber) {
    return (dispatch) => {
      dispatch({ type: constants.FETCH_USER_FEED });

      FeedAPI.fetchUserFeed(userId = null,userLocation=null, pageNumber=1).then(
        (feedItems) => dispatch({ type: constants.RECEIVE_USER_FEED, success: true, payload: feedItems }),
        (error) => dispatch({ type: constants.RECEIVE_USER_FEED, success: false, error })
      );
    }
  }

  static reportItem(itemId, itemType, userId) {

  }

  static showItemDetail(itemId, itemType, userId) {

  }

  static favoriteItem(itemId, itemType, userId) {

  }

  static fetchSimilarItemsByTag(itemId, itemType, userId) {

  }

  static fetchSimilarItemsByLocation(itemId, itemType, userId) {

  }

  static fetchSimilarItemsByUser(itemId, itemType, userId) {

  }
}
