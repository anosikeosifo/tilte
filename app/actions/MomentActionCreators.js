import { MOMENT_ACTIONS } from '../constants';
import { likeMomentAPI, fetchMomentAPI } from '../api/MomentAPI';
import { buildRequestParams } from '../helpers/HttpHelper';


const momentActionSuccess = (actionType, payload) => ({
  type: MOMENT_ACTIONS[`${actionType}Success`], success: true, payload 
});

const momentActionError = (actionType, error) => ({
  type: MOMENT_ACTIONS[`${actionType}Error`], success: false, error 
});

const triggerlikeMoment = () => ({
  type: MOMENT_ACTIONS.likeRequest
});

export const likeMoment = (momentId, actorId) => {
  const actionType = 'like';

  const requestData = buildRequestParams({ 
    moment_id: momentId, user_id: actorId 
  });
  
  return (dispatch) =>  {
    triggerlikeMoment();

    likeMomentAPI(requestData)
      .then(payload => dispatch(momentActionSuccess(actionType, payload)))
      .catch(error => dispatch(momentActionError(actionType, error)));
  }
};

const triggerFetchMomentDetails = (dispatch) => ({
  type: MOMENT_ACTIONS.fetchDetails
});

export const fetchMomentDetails = (dispatch) => {
  const actionType = 'fetchDetails';

  const requestData = buildRequestParams({ 
    moment_id: momentId, user_id: actorId 
  });
  
  return (dispatch) =>  {
    triggerFetchMomentDetails();

    fetchMomentAPI(requestData)
      .then(payload => dispatch(momentActionSuccess(actionType, payload)))
      .catch(error => dispatch(momentActionError(actionType, error)));
  }
};