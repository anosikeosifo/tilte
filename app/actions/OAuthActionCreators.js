import { OAUTH_ACTIONS } from '../constants';
import { oauthLoginAPI, verifyOAuthAPI } from '../api/OAuthAPI';
import { setupFacebook, loginWithFacebook } from '../helpers/OAuthHelper';


const oauthActionSuccess = (actionType, payload) => ({
  type: OAUTH_ACTIONS[`${actionType}Success`], success: true, payload
});

const oauthActionError = (actionType, error) => ({
  type: OAUTH_ACTIONS[`${actionType}Error`], success: false, error
});

const triggerAuthentication = (authType) => ({
  type: `${ authType }Request`
});

const setAuthenticatedUser = (authType) => ({
  type: `${ authType }Request`
});

const triggerOAuthVerification = (authType) => ({
  type: `${ authType }VerifyRequest`
});

const handlefacbookLoginCheck = (dispatch, response) => {
  switch (response.data.status) {
    case 'not_authorized':
    case 'unknown':
      setGuestUser(dispatch);
      break;
    case 'connected':
      handleOAuthVerification(dispatch, response.data, 'facebookAuth');
      break;
    default:
  }
};

const setGuestUser = (dispatch) => dispatch({
  type: OAUTH_ACTIONS.setGuestUser
});


const handleOAuthVerification = (dispatch, response, authType) => {
  const actionType = `${ authType }Verify`;

  triggerOAuthVerification(authType);
  verifyOAuthAPI(authType, response)
    .then(payload => dispatch(oauthActionSuccess(actionType, payload)))
    .then(error => dispatch(oauthActionError(actionType, error)));
}

export const checkLoggedInStatus = () => {
  return (dispatch) => {
    setupFacebook('1344401955668048', (response) => {
      handlefacbookLoginCheck(dispatch, response)
    });
  }
};

const handleFailedLogin = (failureMessage) => {
  console.log('fb login failed: ', failureMessage);
}

export const triggerFacebookLogin = (dispatch, response) => {
  const actionType = 'facebookAuth';

  return (dispatch) => {
    loginWithFacebook((response) => {
      if(!response.success) return handleFailedLogin(response.message);

      oauthLoginAPI(actionType, response.data)
        .then(payload => dispatch(oauthActionSuccess(actionType, payload)))
        .catch(error => dispatch(oauthActionError(actionType, error)));
    });
  }
};

export const triggerTwitterLogin = () => {
  const actionType = 'twitterAuth';
}

export const triggerLinkedinLogin = () => {
  const actionType = 'linkedinAuth';
}
