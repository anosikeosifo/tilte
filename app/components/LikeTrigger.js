import '../styles/like_trigger.scss';

import React, { Component, PropTypes } from 'react';
import { ICON_FAVORITE } from 'constants';
import classNames from 'classnames';
import Icon from './Icon';
import update from 'react-addons-update';

class LikeTrigger extends Component {
  constructor() {
    super();

    this.state = {
      color: {
        active: '#ffa600',
        default: '#c1c1c1',
      },
      iconColor: null,
    }
  }

  componentWillMount() {
    this.setState(Object.assign({}, this.state, { 
      iconColor: this.getIconColor(this.props.activeState)
    }));
  }

  getIconColor(interactionState) {
    return this.state.color[interactionState];
  }

  handleMouseEnter() {
    if(this.props.isEnabled) {
      this.setState(update(this.state, { 
        iconColor: { $set: this.getIconColor("active") }
      }));
    }
  }

  handleMouseLeave() {
    if(this.props.isEnabled) {
      this.setState(update(this.state, { 
        iconColor: { $set: this.getIconColor("default") }
      }));
    }
  }

  handleClick() {
    if(this.props.isEnabled) this.props.likeAction(this.props.momentId, "1");
  }

  render() {
    let triggerDOM = null;

    if (this.props.isEnabled) {
      triggerDOM = (<section className='component__like__triger'
         onMouseEnter={ this.handleMouseEnter.bind(this) } 
         onMouseLeave={ this.handleMouseLeave.bind(this) }
         onClick={ this.handleClick.bind(this) }>
          <Icon icon={ICON_FAVORITE} color={ this.state.iconColor } />
        </section>)
    } else {
      triggerDOM = (
        <section className='component__like__triger'>
          <Icon icon={ICON_FAVORITE} color={ this.state.iconColor } />
        </section>
      );
    }

    return triggerDOM;
  }
}

LikeTrigger.PropTypes = {
  user: PropTypes.object
}

export default LikeTrigger;
