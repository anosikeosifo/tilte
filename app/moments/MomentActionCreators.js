import { MOMENT_ACTIONS } from './constants';
import { likeMomentAPI, fetchMomentAPI, fetchCommentsAPI, postCommentAPI ,loadMapAPI, fetchTopMomentsAPI } from './MomentAPI';
import { buildUrlQueryParams } from '../helpers/HttpHelper';

const momentActionSuccess = (actionType, payload) => ({
  type: MOMENT_ACTIONS[`${actionType}Success`], success: true, payload
});

const momentActionError = (actionType, error) => ({
  type: MOMENT_ACTIONS[`${actionType}Error`], success: false, error
});

const triggerlikeMoment = () => ({
  type: MOMENT_ACTIONS.likeRequest
});

const triggerFetchMomentDetails = () => ({
  type: MOMENT_ACTIONS.fetchDetails
});

const triggerFetchComments = () => ({
  type: MOMENT_ACTIONS.fetchCommentsRequest
});

const triggerPostComment = () => ({
  type: MOMENT_ACTIONS.postCommentRequest
});

const triggerSaveMoment = () => ({
  type: MOMENT_ACTIONS.saveMomentRequest
});

const triggerFetchTopMoments = () => ({
  type: MOMENT_ACTIONS.fetchTopMomentsRequest
});

export const likeMoment = (eventId, actorId) => {
  const actionType = 'like';
  const requestParams = {
    event_id: eventId, user_id: actorId
  };

  return (dispatch) => {
    triggerlikeMoment();

    likeMomentAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      .catch(error => dispatch(eventActionError(actionType, error)));
  }
};

export const fetchMomentDetails = (eventId, actorId) => {
  const actionType = 'fetchDetails';
  const requestParams = buildUrlQueryParams({
    id: eventId, user_id: actorId
  });

  return (dispatch) => {
    triggerFetchMomentDetails();

    //this should take requestParams as arg
    fetchMomentAPI(eventId)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(actionType, error)));
  }
};

export const fetchComments = (eventId, actorId) => {
  const actionType = 'fetchComments';
  const requestParams = {
    event_id: eventId, user_id: actorId, page: '1'
  };

  return (dispatch) => {
    triggerFetchComments();
    fetchCommentsAPI (requestParams)
      .then(payload => eventActionSuccess(actionType, payload))
      .catch(error => eventActionError(actionType, error));
  };
}

export const postComment = (eventId, actorId, content) => {
  const actionType = 'postComment';
  const requestParams = {
    event_id: eventId,
    user_id: actorId,
    text: content,
  };

  return (dispatch) => {
    triggerPostComment();
    postCommentAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(error)));
  };
};

export const loadMapView = (eventId, actorId, longitude=null, latitude=null) => {
  const actionType = 'loadMap';
  const requestParams = {
    event_id: eventId,
    user_id: actorId,
    longitude,
    latitude,
  };

  return (dispatch) => {
    triggerLoadMap();
    loadMapAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(error)));
  };
}

export const saveMoment = (eventId, actorId) => {
  const actionType = 'saveMoment';
  const requestParams = {
    event_id: eventId,
    user_id: actorId,
  };

  return (dispatch) => {
    triggerSaveMoment();
    saveMomentAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(error)));
  };
}

export const fetchTopMoments = (userId) => {
  const actionType = 'fetchTopPosts';
  const requestParams = { userId };

  return (dispatch) => {
    triggerFetchTopMoments();
    fetchTopMomentsAPI(requestParams)
      .then(payload => dispatch(eventActionSuccess(actionType, payload)))
      // .catch(error => dispatch(eventActionError(error)));
  };
}
