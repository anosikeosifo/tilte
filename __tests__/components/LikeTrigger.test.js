
import React from 'react';
import renderer from 'react-test-renderer';
import LikeTrigger from '../../app/components/LikeTrigger';
import { mount } from 'enzyme';
import { likeMoment } from '../../app/actions/MomentActionCreators';
 

describe('LikeTrigger Component', () => {
  it('should match the snapshot for initial state', () => {
    const tree = renderer.create(
      <LikeTrigger/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  }); 

  it('should contain an icon ', () => {
    const tree = renderer.create(
      <LikeTrigger/>
    ).toJSON();

    const icon = tree.children.filter(child => child.type === 'svg')[0];
    expect(icon).toBeDefined();
  });

  describe('When enabled', () => {
    const component = renderer.create(
      <LikeTrigger isEnabled={ true } uiState='default' likeAction={ likeMoment }/>
    ).toJSON();

    it('Should change icon color on hover', () => {
      component.props.onMouseEnter();
      component.props.onMouseLeave();
      expect(component).toMatchSnapshot();
    });

    it('Should change icon color on click', () => {
      component.props.onClick();
      expect(component).toMatchSnapshot();
    });
  });

  describe('When disabled', () => {
    it('Should not have UI interactions', () => {
      const component = renderer.create(
        <LikeTrigger isEnabled={ false } uiState='default'/>
      ).toJSON();

      expect(component.props.onMouseEnter).not.toBeDefined();
      expect(component.props.onMouseLeave).not.toBeDefined();
      expect(component.props.onClick).not.toBeDefined();
      expect(component).toMatchSnapshot();
    });
  });
});