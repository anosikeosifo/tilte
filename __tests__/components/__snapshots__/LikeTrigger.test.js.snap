exports[`LikeTrigger Component When disabled Should not have UI interactions 1`] = `
<section
  className="component__like__triger">
  <svg
    height="24px"
    style={
      Object {
        "display": "inline-block",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 1024 1024"
    width="24px">
    <path
      d={undefined}
      style={
        Object {
          "fill": undefined,
        }
      } />
  </svg>
</section>
`;

exports[`LikeTrigger Component When enabled Should change icon color on click 1`] = `
<section
  className="component__like__triger"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  <svg
    height="24px"
    style={
      Object {
        "display": "inline-block",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 1024 1024"
    width="24px">
    <path
      d={undefined}
      style={
        Object {
          "fill": undefined,
        }
      } />
  </svg>
</section>
`;

exports[`LikeTrigger Component When enabled Should change icon color on hover 1`] = `
<section
  className="component__like__triger"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  <svg
    height="24px"
    style={
      Object {
        "display": "inline-block",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 1024 1024"
    width="24px">
    <path
      d={undefined}
      style={
        Object {
          "fill": undefined,
        }
      } />
  </svg>
</section>
`;

exports[`LikeTrigger Component should match the snapshot for initial state 1`] = `
<section
  className="component__like__triger">
  <svg
    height="24px"
    style={
      Object {
        "display": "inline-block",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 1024 1024"
    width="24px">
    <path
      d={undefined}
      style={
        Object {
          "fill": undefined,
        }
      } />
  </svg>
</section>
`;
